### PR TITLE
GSoC - Compiling the Java tutorials codes using Apache Ant.

### PR DESCRIPTION
**Pull request by Cartucho**
Adding to samples/java/tutorial:
* CMakeLists.txt
* build.xml

Go through the Java tutorials directories and **compile** the **.java** files.

### CODE REVIEW

File: samples/CMakeLists.txt [hunk 1]
```diff
@@ -82,6 +83,8 @@ add_subdirectory(cpp)
 add_subdirectory(dnn)
 # FIXIT: can't use cvconfig.h in samples: add_subdirectory(gpu)
 
+add_subdirectory(java/tutorial_code)
+
```
Comment (alalek) [inline samples/CMakeLists.txt]: Please remove this. It doesn't work well for samples standalone build (build samples only against OpenCV installation or build directory).
Anyway ANT_EXECUTABLE/OpenCV_BINARY_DIR are not defined for these builds, and location of opencv.jar is not valid (winpack puts "jar" file under "java" subdirectory, not "bin").

File: samples/java/tutorial_code/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# ----------------------------------------------------------------------------
+#  CMake file for Java tutorials compilation.
+#
+# ----------------------------------------------------------------------------
+if(NOT ANT_EXECUTABLE)
```
Comment (alalek) [inline samples/java/tutorial_code/CMakeLists.txt]: `... OR NOT TARGET opencv_java)`

**Comment by alalek** [comment]
:+1:

**Comment by Cartucho** [comment]
screenshot from 2017-06-28 12-17-26

**Pull request merged**